Build: Include prerelease install info in release blog post

platinumazure opened the pull request:
<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
-->

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Build improvement to ensure that prerelease blog posts contain installation and migration guide information.

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

**What changes did you make? (Give an overview)**

Added some logic to ensure that if we are doing a prerelease, that the info about installation and migration guide be included to reduce manual edits after the release.

Caveat: The release type in the intro paragraph is calculated in eslint-release based solely on the commit log. I'm hoping to submit a separate PR in the near future to fix that by respecting if the current version (as of before the release process) is also a prerelease.

**Is there anything you'd like reviewers to focus on?**

I don't know how to test this (either manually or via unit tests). I'd be grateful for pointers. (Leaving as WIP/do not merge to avoid accidental merging.)

platinumazure commented:
@eslint/eslint-team Would be great to merge this in before next release. Could I get some tips on how to test this?

ilyavolodin commented:
We currently don't have any tests for Makefile. There were couple of attempts before to do that, but none of them were complete. It would be great to test Makefile and release process, but I think you will have to extensively mock file system and shelljs to do that correctly.

ilyavolodin commented on templates/blogpost.md.ejs:
```diff
@@ -27,6 +27,31 @@ function outputList(items) {
 
 %>
 
+<% if (prereleaseMajorVersion) { %>
```
Am I not understanding this change correctly, or based on the changes in Makefile, `prereleaseMajorVersion` will always have some sort of value?

platinumazure replied:
You're absolutely correct, I forgot to compare the versions in the Makefile and only hydrate the context object when we're in a pre-release. I'll need to fix that. Thanks!

platinumazure commented:
Hi @ilyavolodin, I've fixed the issue you called out. What I've done is a bit weird so I also added some comments (which also contain potential test cases if we ever figure out a way to unit test the Makefile). Thanks again for taking a look!

ilyavolodin commented on Makefile.js:
```diff
@@ -306,11 +306,29 @@ function release(ciRelease) {
+     * Premajor release should have identical "next major version".
+     * Preminor and prepatch release will not.
+     * 5.0.0-alpha.0 --> next major = 5, current major = 5
+     * 4.4.0-alpha.0 --> next major = 5, current major = 4
```
I'm a bit confused. When would this be the case? Do we even need to check for this?

platinumazure replied:
No, we don't normally do preminor and prepatch releases, but it's theoretically possible; and in those cases, the code I had before would misbehave (it would show major-release migration guide and installation information when we're not on a premajor release).

platinumazure commented:
Leaving this open another day or so in case more team members want to review.

platinumazure commented:
I've thought about it some more. I think we should *not* merge this for 5.0.0, because that's not going to be a prerelease. Thus, this PR doesn't add value to that release, but does add risk.

Re-adding "do not merge". Let's put this in after the major release.

platinumazure commented:
@eslint/eslint-team I would love another review or two on this before I merge later this week. Thanks!
